go/private/rules/cgo.bzl

```diff
@@ -176,35 +176,49 @@ def _cgo_codegen_impl(ctx):
-      # us the static variant. We'll get one file for each transitive dependency,
-      # so the same file may appear more than once.
-      if (lib.basename.startswith("lib") and
-          any([lib.basename.endswith(ext) for ext in SHARED_LIB_EXTENSIONS])):
-        # If the loader would be able to find the library using rpaths,
-        # use -L and -l instead of hard coding the path to the library in
-        # the binary. This gives users more flexibility. The linker will add
-        # rpaths later. We can't add them here because they are relative to
-        # the binary location, and we don't know where that is.
-        libname = lib.basename[len("lib"):lib.basename.rindex(".")]
-        linkopts.extend(['-L', lib.dirname, '-l', libname])
-      else:
-        linkopts.append(lib.path)
-    linkopts.extend(d.cc.link_flags)
+    if hasattr(d, "cc"):
+      inputs = sets.union(inputs, d.cc.transitive_headers)
+      deps = sets.union(deps, d.cc.libs)
+      cppopts.extend(['-D' + define for define in d.cc.defines])
+      for inc in d.cc.include_directories:
+        _include_unique(cppopts, "-I", inc, seen_includes)
+      for inc in d.cc.quote_include_directories:
+        _include_unique(cppopts, "-iquote", inc, seen_quote_includes)
+      for inc in d.cc.system_include_directories:
+        _include_unique(cppopts, "-isystem", inc, seen_system_includes)
+      for lib in as_iterable(d.cc.libs):
+        # If both static and dynamic variants are available, Bazel will only give
+        # us the static variant. We'll get one file for each transitive dependency,
+        # so the same file may appear more than once.
+        if (lib.basename.startswith("lib") and
+            any([lib.basename.endswith(ext) for ext in SHARED_LIB_EXTENSIONS])):
+          # If the loader would be able to find the library using rpaths,
+          # use -L and -l instead of hard coding the path to the library in
+          # the binary. This gives users more flexibility. The linker will add
+          # rpaths later. We can't add them here because they are relative to
+          # the binary location, and we don't know where that is.
+          libname = lib.basename[len("lib"):lib.basename.rindex(".")]
+          linkopts.extend(['-L', lib.dirname, '-l', libname])
+        else:
+          linkopts.append(lib.path)
+      linkopts.extend(d.cc.link_flags)
+    elif hasattr(d, "objc"):
+      cppopts.extend(['-D' + define for define in d.objc.define.to_list()])
+      for inc in d.objc.include:
+        _include_unique(cppopts, "-I", inc, seen_includes)
+      for inc in d.objc.iquote:
+        _include_unique(cppopts, "-iquote", inc, seen_quote_includes)
+      for inc in d.objc.include_system:
+        _include_unique(cppopts, "-isystem", inc, seen_system_includes)
+      # TODO(jayconrod): do we need to link against dynamic libraries or
+      # frameworks? We link against *_fully_linked.a, so maybe not?
```

Comment:
sdk_framework would be great

Reply:
Also if modules are enabled you might want to add -fmodules

Reply:
Would you be willing to do a PR to add these? I'm not familiar enough with Objective C to know if I'm doing it right. :)

Reply:
i'll try ! thanks for doing this :D